web/src/components/FeedbackModal/FeedbackModal.test.tsx

```diff
@@ -0,0 +1,294 @@
+    });
+
+    it("does not submit reported issue when issue summary input is empty and shows inline validation error", () => {
+      renderFeedbackModal();
+
+      fireEvent.click(screen.getByText(Config.feedbackModal.betaModalBugReportButtonText));
+      fireEvent.change(screen.getByLabelText("Issue summary"), { target: { value: "random summary text" } });
+      fireEvent.click(screen.getByText(Config.feedbackModal.feedbackModalSubmitButtonText));
+
+      expect(screen.getByText(Config.feedbackModal.feedbackModalInlineErrorText)).toBeInTheDocument();
+      expect(mockApi.postIssue).not.toHaveBeenCalled();
+    });
+
+    it("does not submit reported issue when issue detail input is empty and shows inline validation error", () => {
+      renderFeedbackModal();
+
+      fireEvent.click(screen.getByText(Config.feedbackModal.betaModalBugReportButtonText));
+      fireEvent.change(screen.getByLabelText("Issue details"), { target: { value: "random details text" } });
+      fireEvent.click(screen.getByText(Config.feedbackModal.feedbackModalSubmitButtonText));
+
+      expect(screen.getByText(Config.feedbackModal.feedbackModalInlineErrorText)).toBeInTheDocument();
+      expect(mockApi.postIssue).not.toHaveBeenCalled();
+    });
+
+    it("opens issues dialog when isBug prop is used and then returns to feedback selection dialog after issue submission", async () => {
+      renderFeedbackModal(true);
+
+      fireEvent.change(screen.getByLabelText("Issue summary"), { target: { value: "random summary text" } });
+      fireEvent.change(screen.getByLabelText("Issue details"), { target: { value: "random details text" } });
+      fireEvent.click(screen.getByText(Config.feedbackModal.feedbackModalSubmitButtonText));
+      await waitFor(() => {
+        screen.getByText(Config.feedbackModal.successfulSubmissionModalButtonText);
+      });
+      fireEvent.click(screen.getByText(Config.feedbackModal.successfulSubmissionModalButtonText));
+
+      await waitFor(() => {
+        expect(screen.getByText(Config.feedbackModal.betaModalTitle)).toBeInTheDocument();
+      });
+    });
+
+    it("renders successful submission modal after reported issed is submitted", async () => {
+      global.innerWidth = 500;
+
+      const userAgent = {
+        os: { name: "Mac OS", version: "10" },
+        device: { vendor: "Google", model: "Pixel", type: "Mobile" },
+        browser: { name: "Firefox", version: "6.5", major: "6" },
+      };
+      getResult.mockReturnValue(generateParsedUserAgent(userAgent));
+
+      const userData = generateUserData({});
+      useMockUserData(userData);
+
+      const issue = generateIssueRequest({
+        context: "random summary text",
+        browser: "Firefox v.6.5",
+        device: "Mac OS 10 Google Pixel Mobile",
+        screenWidth: "500 px",
+        detail: "random details text",
+        pageOfRequest: "roadmap/test",
+      });
```

Comment:
I wouldn't recommend using a generator for an object that you're using as an expectation. This is the `expectedIssue` that you expect something to have been called with. The thing about a generator is that it will fill in random values for anything you don't provide.

If we add a new field to the issue object, this code would still compile and it would have a random value floating around in there that may or may not be correct. Whereas if you build the expected object explicitly here like:
```
const expectedIssue: IssueRequest = {
... all the stuff here
}
```
then it would fail to compile if we add a new field, which is good, because it would force us to update the test and be clear about what we expect to happen to this field in the situation.